ksonnet-gen/astext/astext.go

```diff
@@ -34,7 +34,7 @@ type Object struct {
 
 var (
 	// reFieldStr matches a field id that should be enclosed in quotes.
-	reFieldStr = regexp.MustCompile(`^[_A-Za-z]+[_A-Za-z0-9\-]*$`)
+	reFieldStr = regexp.MustCompile(`^([_A-Za-z0-9\.]?[A-Za-z0-9\-_\.]+(\.[A-Za-z0-9\-_]+)*)?$`)
 	// reField matches a field id.
 	reField = regexp.MustCompile(`^[_A-Za-z]+[_A-Za-z0-9]*$`)
```

Comment:
This doesn't take keywords into account like we did in `printer.shouldUnquoteFieldID()` - I believe they should be caught during the printing phase, but I will do a bit of further exploration whether it is even a valid AST to create an `ObjectFieldID` with an invalid identifier.

Reply:
If end up needing to, we should refactor to share that logic.